### PR TITLE
Support array datatype in PostgreSQL

### PR DESCRIPTION
**Pull request by billy1624**
## PR Info

- Closes https://github.com/SeaQL/sea-orm/issues/576

## Adds

- [x] `postgres-array` feature
- [x] `ColumnType::Array` variant
- [x] Conversion between `sea_query::ColumnType::Array` and `sea_orm::ColumnType::Array`
- [x] Support `Vec<T>` to be used as model fields in PostgreSQL

### CODE REVIEW

File: src/entity/column.rs [hunk 1]
```diff
@@ -432,6 +434,66 @@ impl From<ColumnType> for sea_query::ColumnType {
...
+                            variants: variants.clone(),
+                        },
+                        ColumnType::Array(column_type) => {
+                            from_rc_box_column_type(column_type.clone())
+                        }
+                    }
+                }
+                // Reconstruct it as `sea_query::ColumnType::Array`
+                let column_type = from_rc_box_column_type(column_type);
+                sea_query::ColumnType::Array(SeaRc::new(Box::new(column_type)))
```
Comment (billy1624) [inline src/entity/column.rs:495]: Convert `sea_orm::ColumnType` into `sea_query::ColumnType`

File: src/entity/column.rs [hunk 2]
```diff
@@ -470,6 +532,64 @@ impl From<sea_query::ColumnType> for ColumnType {
...
+                        },
+                        sea_query::ColumnType::Array(column_type) => {
+                            from_rc_box_column_type(column_type.clone())
+                        }
+                        _ => unimplemented!(),
+                    }
+                }
+                // Reconstruct it as `ColumnType::Array`
+                let column_type = from_rc_box_column_type(column_type);
+                Self::Array(SeaRc::new(Box::new(column_type)))
```
Comment (billy1624) [inline src/entity/column.rs:591]: Convert `sea_query::ColumnType` into `sea_orm::ColumnType`

**Review by billy1624 — COMMENTED** [review]
These two conversion look ugly. But I cannot convert it with `Into<T>`. Because the `ColumnType` is behind a `Rc` / `Arc` so I cannot take ownership of the `ColumnType`, that blocking me from using `Into<T>`.

Maybe there is another way around it. Any suggestion?

**Comment by billy1624** [comment]
Hey @tyt2y3, ready for review. Do you think we need to support entity generation of PostgreSQL's array datatypes?

**Comment by billy1624** [comment]
Okay, I managed to refactor it @tyt2y3

**Comment by ikrivosheev** [comment]
@tyt2y3 hello! Where do you find the sea-schema package with version: 0.10.1?)

**Comment by tyt2y3** [comment]
I just released it

**Pull request merged**